mos_tests/nova/nova_test.py

```diff
@@ -601,3 +606,80 @@ def test_boot_instance_from_volume_bigger_than_flavor_size(self):
         # Check that instance is reachable
         ping = common_functions.ping_command(self.floating_ip.ip)
         self.assertTrue(ping, "Instance after creation is not reachable")
+
+
+class TestNovaDeferredDelete(TestBase):
+    """Nova Deferred Delete test cases"""
+
+    @classmethod
+    @pytest.fixture(scope='class', autouse=True)
```

Comment:
> Also if your tests is undestructive - you should delete created security group

It will be done in `os_conn.cleanup_network()`.

```
def teardown(cls, os_conn):
	yield
	os_conn.cleanup_network()

Will perform:
    self.delete_security_groups()
	self.delete_keypairs()
    self.delete_floating_ips()
    self.delete_servers()
    self.delete_ports(networks)
    self.delete_subnets(networks)
    self.delete_routers()
```